lib/Cleantalk/ApbctWP/Honeypot.php

```diff
@@ -0,0 +1,164 @@
+<?php
+
+namespace Cleantalk\ApbctWP;
+
+use Cleantalk\Common\TT;
+use Cleantalk\ApbctWP\Variables\Post;
+use Cleantalk\ApbctWP\Variables\AltSessions;
+
+class Honeypot
+{
+    /**
+     * Returns HTML of a honeypot hidden field to the form. If $form_method is GET, adds a hidden submit button.
+     * @param $form_type
+     * @param string $form_method
+     * @return string
+     */
+    public static function generateHoneypotField($form_type, $form_method = 'post')
+    {
+        global $apbct;
+
+        if ( ! $apbct->settings['data__honeypot_field'] || apbct_exclusions_check__url() || apbct_is_amp_request()) {
+            return '';
+        }
+
+        $apbct_event_id = mt_rand(0, 100000);
+
+        // field label (preventing validators warning)
+        $label = '<label ' .  'class="apbct_special_field" ' .  'id="apbct_label_id' . $apbct_event_id . '" ' .
+            'for="apbct__email_id__' . $form_type . '_' . $apbct_event_id . '"' .  '>' . $apbct_event_id . '</label>';
+
+        // generate the honeypot trap input
+        $honeypot = $label . '<input
+            id="apbct__email_id__' . $form_type . '_' . $apbct_event_id . '"
+            class="apbct_special_field apbct__email_id__' . $form_type . '"
+            name="apbct__email_id__' . $form_type . '_' . $apbct_event_id . '"
+            type="text" size="30" maxlength="200" autocomplete="off"
+            value="' . $apbct_event_id . '" apbct_event_id="' . $apbct_event_id . '"
+            />';
+
+        // if POST, add a hidden input to transfer apbct_event_id to the form data
+        if ($form_method === 'post') {
+            return $honeypot . '<input id="apbct_event_id_' . $form_type . '_' . $apbct_event_id . '"
+                class="apbct_special_field" name="apbct_event_id" type="hidden" value="' . $apbct_event_id . '" />';
+        }
+
+        // if GET, place a submit button if method is get to prevent keyboard send misfunction
+        return $honeypot . '<input id="apbct_submit_id__' . $form_type . '_' . $apbct_event_id . '" 
```

Comment:
For the GET forms need to add honeypot hidden input too. But not for the all forms, only for the integrated ones (search form for example). We have to discuss this.

Reply:
There will be separate task.